DSCResources/MSFT_xDnsConnectionSuffix/MSFT_xDnsConnectionSuffix.psm1

```diff
@@ -0,0 +1,140 @@
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData -StringData @'
+PropertyMismatch          = Property '{0}' does NOT match. Expected '{1}', actual '{2}'.
+CheckingConnectionSuffix  = Checking connection suffix matches '{0}'.
+ResourceInDesiredState    = Resource is in the desired state.
+ResourceNotInDesiredState = Resource is NOT in the desired state.
+SettingConnectionSuffix   = Setting connection suffix '{0}' on interface '{1}'.
+RemovingConnectionSuffix  = Removing connection suffix '{0}' on interface '{1}'.
+'@
+}
+
+function Get-TargetResource {
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String] $InterfaceAlias,
+        
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String] $ConnectionSpecificSuffix,
+        
+        [Parameter()]
+        [System.Boolean] $RegisterThisConnectionsAddress = $true,
+
+        [Parameter()]
+        [System.Boolean] $UseSuffixWhenRegistering = $false, 
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [System.String] $Ensure = 'Present'
+    )
+    $dnsClient = Get-DnsClient -InterfaceAlias $InterfaceAlias -ErrorAction SilentlyContinue;
+    $targetResource = @{
+        InterfaceAlias = $dnsClient.InterfaceAlias;
+        DnsSuffix = $dnsClient.ConnectionSpecificSuffix;
+        RegisterThisConnectionsAddress = $dnsClient.RegisterThisConnectionsAddress;
+        UseSuffixWhenRegistering = $dnsClient.UseSuffixWhenRegistering;
+    }
+    if ($Ensure -eq 'Present') {
+        ## Test to see if the connection-specific suffix matches
+        Write-Verbose ($LocalizedData.CheckingConnectionSuffix -f $ConnectionSpecificSuffix);
+        $targetResource['Ensure'] = if ($dnsClient.ConnectionSpecificSuffix -eq $ConnectionSpecificSuffix) { 'Present' } else { 'Absent' }
```

Comment:
nit: This goes over our 100 line limit, and the style is to put curly braces on their own lines when we can.

ie:
````powershell
$targetResource['Ensure'] = if ($dnsClient.ConnectionSpecificSuffix -eq $ConnectionSpecificSuffix)
    {
        'Present'
    }
    else
    {
        'Absent'
    }
````